src/protocol/libp2p/ping/mod.rs

```diff
@@ -192,6 +195,19 @@ impl Ping {
                     Some(_) => {}
                     None => return,
                 },
+                _ = interval.tick() => {
+                    for peer in &self.peers {
+                        tracing::trace!(target: LOG_TARGET, ?peer, "sending ping");
+                        if let Err(error) = self.service.open_substream(*peer) {
```

Comment:
The libp2p ping protocol spec states that the repeated pings should go on the same substream: https://github.com/libp2p/specs/blob/master/ping/ping.md

Because of this, the issue is actually more difficult due to the need to not keep the connection alive by this ping substream.

I don't think we should break the spec even if it works with the current libp2p implementation. @lexnv what do you think?